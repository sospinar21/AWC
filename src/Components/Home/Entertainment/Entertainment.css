.sidebar {
  width: 24%;
  height: 99vh;
  background-color: white;
  overflow: scroll;
  box-shadow: 1px 4px 11px lightgrey; }

.entertainment {
  width: 100%;
  height: 100vh; }

.navBar-entert {
  display: flex; }

.videos {
  width: 100%;
  margin-left: 11vw; }
