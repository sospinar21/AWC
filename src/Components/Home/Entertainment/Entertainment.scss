.sidebar {
  width: 24%;
  height: 99vh;
  background-color: white;
  overflow: scroll;
  box-shadow: 1px 4px 11px lightgrey;
}