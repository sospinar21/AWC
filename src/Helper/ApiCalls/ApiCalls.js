import React, {Component} from 'react';

class ApiCalls extends Component {
  constructor(){
    super()
  }

  fetchEvents = async () => {
    const url = `https://api.awc.dance/events?city=vermont`;
    const response = await fetch(url)
    const data= await response.json();
    return data;
  } 

  fetchStudios = () => {
    const url = `https://api.awc.dance/?city=80202`;
    return fetch(url, {
      method: 'GET',
      headers: {
        'Content-type': 'application/json',
      }
    }).then( response => response.json() )
      .then( data => data )   
  }

  fetchVideos= async () => {
    var url = 'https://api.awc.dance/youtubevideos'
    const response = await fetch(url)
    const data = await response.json()
    return data
  }
}

export default ApiCalls;