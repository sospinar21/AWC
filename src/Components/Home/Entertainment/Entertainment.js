import React, { Component } from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import { NavBar } from '../../Home/NavBar/NavBar';
import './Entertainment.css';
import Videos from '../Videos/Videos';
import ApiCalls from '../../../Helper/ApiCalls/ApiCalls';

export class Main extends Component {
  constructor() {
    super();

    this.state = {
      selectedVideo: ''
    };
  }

  render () {

    return (
      <div>
        <div className='entertainment'>
          <div className='navBar-entert'>
            <NavBar />
            <div className='videos'>
              <Videos />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export const mapStateToProps = (state) => {
  return ({
    suggestedEvents: state.suggestedEvents
  });
};

export const mapDispatchToProps = dispatch => ({
  
});

Main.propTypes = {
  suggestedEvents: PropTypes.array
};

export default connect(mapStateToProps, mapDispatchToProps)(Main);