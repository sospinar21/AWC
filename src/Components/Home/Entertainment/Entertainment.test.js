import { Entertainment, mapStateToProps } from '../Entertainment/Entertainment'
import React from 'react';
import ReactDOM from 'react-dom';
import { mount, shallow } from 'enzyme';
import  ApiCalls  from '../../../Helper/ApiCalls/ApiCalls';


describe('entertaiment', () => {
  let entertaiment;
  let mockProps;

  beforeEach(() => {

    mockProps = {
      suggestedEvents: [{name: 'party'}],
    }

    entertaiment = shallow(<Entertainment {...mockProps} /> )

  })

  it('matches snapshot', () => {
    expect(entertaiment).toMatchSnapshot();
  })

  describe('mapStateToProps', () => {
    
    let suggestedEvents;
    let mappedProps

    it('returns an object with the suggestedEvents', () => {

      const mockState = {
        suggestedEvents: {name: 'Workshop'}
      }

      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(mockState);
      
    })
  })
})
