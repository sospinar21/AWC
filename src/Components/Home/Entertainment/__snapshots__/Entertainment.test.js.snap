// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`entertaiment matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Entertainment
    suggestedEvents={
        Array [
            Object {
              "name": "party",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="entertainment"
>
        <div
                className="navBar-entert"
        >
                <NavBar />
                <div
                        className="videos"
                >
                        <Connect(Videos) />
                </div>
        </div>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="navBar-entert"
>
          <NavBar />
          <div
                    className="videos"
          >
                    <Connect(Videos) />
          </div>
</div>,
        "className": "entertainment",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavBar />,
            <div
              className="videos"
>
              <Connect(Videos) />
</div>,
          ],
          "className": "navBar-entert",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Connect(Videos) />,
              "className": "videos",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="entertainment"
>
          <div
                    className="navBar-entert"
          >
                    <NavBar />
                    <div
                              className="videos"
                    >
                              <Connect(Videos) />
                    </div>
          </div>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="navBar-entert"
>
            <NavBar />
            <div
                        className="videos"
            >
                        <Connect(Videos) />
            </div>
</div>,
          "className": "entertainment",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavBar />,
              <div
                className="videos"
>
                <Connect(Videos) />
</div>,
            ],
            "className": "navBar-entert",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(Videos) />,
                "className": "videos",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
